jerry-libc/jerry-libc.c

```diff
@@ -257,17 +261,12 @@ strlen (const char *s)
 int
 rand (void)
 {
-  uint32_t intermediate = libc_random_gen_state[0] ^ (libc_random_gen_state[0] << 11);
-  intermediate ^= intermediate >> 8;
+  uint32_t t = _x ^ (_x << 11);
+  _x = _y;
+  _y = _z;
+  _z = _w;
 
-  libc_random_gen_state[0] = libc_random_gen_state[1];
-  libc_random_gen_state[1] = libc_random_gen_state[2];
-  libc_random_gen_state[2] = libc_random_gen_state[3];
-
-  libc_random_gen_state[3] ^= libc_random_gen_state[3] >> 19;
-  libc_random_gen_state[3] ^= intermediate;
-
-  return (int) (libc_random_gen_state[3] % (RAND_MAX + 1));
+  return (int) (0x7FFFFFFF & (_w = (_w ^ (_w >> 19)) ^ (t ^ (t >> 8))));
```

Comment:
As explained above, this refactoring is also unnecessary, please revert. (Here is also a newly introduced issue: this formula does not take `RAND_MAX` into consideration, which is a libc standard-mandated macro and `rand` must work with that. So, if we changed our `stdlib.h` that would not affect this code. So, please, revert.)